src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
+    private final DateParser dateParser = new DateParser();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startDate = dateParser.date(dateFrom);
+        LocalDate stopDate = dateParser.date(dateTo);
+        int [] salaries = new int[names.length];
+        StringBuilder sb = new StringBuilder();
+        /*
+          Checking every date for every name;
+          Validating name; checking date to fit boundaries; calculating total salary for each date;
+          Filling StringBuilder with every valid entry separately;
+        */
+        for (int i = 0; i < names.length; i++) {
+            for (String datum : data) {
+                if (names[i].equals(datum.split(" ")[1])
```

Comment:
Introduce `" "` as constant

Reply:
do I really have to declare?
PrSF String SPACE_SEPARATOR = " ";
I mean, what else it could be)
I agree to explain what is [0] / [1]... in splitted array, but this above feels strange)